src/ocean/util/UnixSocketCommandInterface.d

```diff
@@ -0,0 +1,161 @@
+/*******************************************************************************
+
+    Utility for handling requests via a unix socket.
+
+    Commands can be registered to the extension and when the command is received
+    the provided delegate will be called. Any provided arguments will be split
+    by the space character and provided as an array.
+
+    Copyright:
+        Copyright (c) 2009-2018 Sociomantic Labs GmbH.
+        All rights reserved.
+
+    License:
+        Boost Software License Version 1.0. See LICENSE_BOOST.txt for details.
+        Alternatively, this file may be distributed under the terms of the Tango
+        3-Clause BSD License (see LICENSE_BSD.txt for details).
+
+*******************************************************************************/
+
+module ocean.util.UnixSocketCommandInterface;
+
+import ocean.transition;
+import ocean.core.Verify;
+
+import ocean.core.Enforce;
+import ocean.io.select.EpollSelectDispatcher;
+
+/// ditto
+public class UnixSocketCommandInterface
+{
+    import core.sys.posix.sys.stat;
+
+    import ocean.core.Buffer;
+    import ocean.core.array.Transformation: filter, split;
+    import ocean.io.select.EpollSelectDispatcher;
+    import ocean.net.server.unix.UnixListener;
+    import ocean.text.convert.Integer;
+    import ocean.text.util.StringC;
+
+    /// Unix listener with custom command handling.
+    private UnixSocketListener!(typeof(this)) unix_listener;
+
+    /// Stores the command arguments split by " ";
+    private cstring[] args_buf;
+
+    /// Alias for our handler delegate.
+    public alias void delegate ( cstring[], void delegate (cstring) ) Handler;
+
+    /// Our registered map of handlers by command.
+    private Handler[istring] handlers;
+
+    /// Path to create the unix socket.
+    private istring path;
+
+    /// Mode to apply to the unix socket after binding
+    private int mode = -1;
+
+
+    /***************************************************************************
+
+        Binds to the path and listens for the new commands
+
+        Params:
+            epoll = Epoll instance.
+            path = path to bind the socket to
+            mode = mode to apply to the socket
```

Comment:
It'd be nice to include a link or documentation for which values are possible and what they mean.

Reply:
I'll explain this better - this is just a normal UNIX permission mode (and I default to the reasonable value - `0600`).